### PR TITLE
chore(deps): upgrade k8s version and client-go

### PR DESCRIPTION
**Pull request by fengshunli**
Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-rollouts/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [ ] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/) with a list of types and scopes found [here](https://github.com/argoproj/argo-rollouts/blob/master/.github/workflows/pr-title-check.yml), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [ ] I've signed my commits with [DCO](https://github.com/argoproj/argoproj)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My builds are green. Try syncing with master if they are not. 
* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-rollouts/blob/master/USERS.md).

### CODE REVIEW

**Comment by fengshunli** [comment]
After upgrading the version to 1.26, the structure of the ingress returned by the api has changed. We can discuss how to modify it soon. I submitted an issue to the community, see https://github.com/kubernetes/api/issues/49 @zachaller

File: go.mod [hunk 1]
```diff
@@ -94,57 +77,62 @@ require (
...
 	github.com/cespare/xxhash/v2 v2.2.0 // indirect
 	github.com/chai2010/gettext-go v1.0.2 // indirect
 	github.com/davecgh/go-spew v1.1.1 // indirect
-	github.com/deepmap/oapi-codegen v1.11.0 // indirect
+	github.com/deepmap/oapi-codegen v1.8.2 // indirect
 	github.com/docker/distribution v2.8.1+incompatible // indirect
-	github.com/emicklei/go-restful/v3 v3.10.1 // indirect
-	github.com/evanphx/json-patch v5.6.0+incompatible // indirect
-	github.com/exponent-io/jsonpath v0.0.0-20210407135951-1de76d718b3f // indirect
+	github.com/emicklei/go-restful/v3 v3.9.0 // indirect
```
Comment (andyliuliming) [inline go.mod:86]: why downgrading?

Comment (andyliuliming) [reply go.mod:86]: and many other modules

**Review by andyliuliming — CHANGES_REQUESTED** [review]

File: go.mod (hunk 1 above)
Comment (fengshunli) [reply go.mod:86]: Indirect dependencies on auto-generated version

**Comment by blkperl** [comment]
Closing as this was fixed in https://github.com/argoproj/argo-rollouts/commit/d378ce76b62180f9c8efc4ccd358263f4171353d

**Pull request closed without merging**